src/welcomePage.ts

```diff
@@ -34,12 +37,44 @@ export class WelcomePage {
             );
             let html = fs.readFileSync(this.context.asAbsolutePath(path.join("resources", "welcome", "index.html")), "utf8");
             html = html.replace(/{{root}}/g, vscode.Uri.file(this.context.asAbsolutePath(".")).with({ scheme: "vscode-resource" }).toString());
+
+            const neverShow = this.context.globalState.get(Constants.WelcomePageNeverShow);
+            html = html.replace("{{Checked}}", neverShow ? "checked" : "");
+
             this.panel.webview.html = html;
             this.panel.onDidDispose(() => {
                 this.panel = undefined;
+                const duration = (new Date().getTime() - startTime.getTime()) / 1000;
+                TelemetryClient.sendEvent(Constants.CloseWelcomePageEvent, { duration: duration.toString() });
+            });
+            this.panel.webview.onDidReceiveMessage((message) => {
+                if (message.href) {
+                    TelemetryClient.sendEvent(Constants.LinkClickEvent, { href: message.href });
+                } else if (message.tab) {
+                    TelemetryClient.sendEvent(Constants.TabClickEvent, { tab: message.tab });
+                } else if (message.neverShow !== undefined) {
+                    this.context.globalState.update(Constants.WelcomePageNeverShow, message.neverShow);
+                }
             });
         } else {
             this.panel.reveal(vscode.ViewColumn.One);
         }
     }
+
+    private needToShow() {
+        const packageJSON = vscode.extensions.getExtension(Constants.ExtensionId).packageJSON;
```

Comment:
We could put packageJSON and extensionVersion into Constants.ts as they appear in two places.